addons/binding/org.openhab.binding.omnilink/ESH-INF/thing/channels.xml

```diff
@@ -0,0 +1,313 @@
+		<state min="0" max="100" pattern="%d %%" />
+	</channel-type>
+	<channel-type id="unit_switch">
+		<item-type>Switch</item-type>
+		<label>Load Switch</label>
+		<description>Toggle Load</description>
+		<category>Switch</category>
+		<state pattern="%s" />
+	</channel-type>
+	<channel-type id="scene_switch">
+		<item-type>Switch</item-type>
+		<label>On/Off</label>
+		<description>Send On/Off to a Scene</description>
+		<category>Switch</category>
+		<state pattern="%s" />
+	</channel-type>
+	<channel-type id="scene_on">
+		<item-type>Switch</item-type>
+		<label>On Scene</label>
+		<description>Turn on On Scene</description>
+		<category>Switch</category>
+		<state pattern="%s" />
+	</channel-type>
+	<channel-type id="scene_off">
+		<item-type>Switch</item-type>
+		<label>Off Scene</label>
+		<description>Turn on Off Scene</description>
+		<category>Switch</category>
+		<state pattern="%s" />
+	</channel-type>
+	<channel-type id="scene_toggle">
+		<item-type>Switch</item-type>
+		<label>Scene Toggle</label>
+		<description>Scene Toggle</description>
+		<category>Toggle</category>
+		<state pattern="%s" />
+	</channel-type>
+	<channel-type id="room_state">
+		<item-type>Number</item-type>
+		<label>State</label>
+		<description>Status of room (0 off, 1 on, 2 scene a, 3 scene b, etc)</description>
+		<state pattern="%s">
+			<options>
+				<option value="0">Off</option>
+				<option value="1">On</option>
+				<option value="3">Scene A</option>
+				<option value="4">Scene B</option>
+				<option value="5">Scene C</option>
+				<option value="6">Scene D</option>
+			</options>
+		</state>
+	</channel-type>
+	<channel-type id="flagvalue">
+		<item-type>Number</item-type>
+		<label>Flag Value</label>
+		<description>Represents value for a flag.</description>
+		<category>Flag Control</category>
+	</channel-type>
+	<channel-type id="flagswitch">
+		<item-type>Switch</item-type>
+		<label>On/Off</label>
+		<description>Flag Switch information</description>
+		<category>Switch</category>
+		<state pattern="%s" />
+	</channel-type>
+
+	<!-- Area channels -->
+
+	<channel-type id="omni_area_alarm">
+		<item-type>Number</item-type>
+		<label>Alarm</label>
+		<description>Represents an area alarm.</description>
+		<category>Alarm</category>
+		<state pattern="%s" readOnly="true">
+			<options>
+				<option value="0">Burglary Alarm</option>
+				<option value="1">Fire Alarm</option>
+				<option value="2">Gas Alarm</option>
+				<option value="3">Auxiliary Alarm</option>
+				<option value="4">Freeze Alarm</option>
+				<option value="5">Water Alarm</option>
+				<option value="6">Duress Alarm</option>
+				<option value="7">Temperature Alarm</option>
+			</options>
+		</state>
+	</channel-type>
+	<channel-type id="omni_area_mode">
+		<item-type>Number</item-type>
+		<label>Security Mode</label>
+		<description>Represents the area security mode.</description>
+		<category>Alarm</category>
+		<state pattern="%s">
+			<options>
```

Comment:
@boc-tothefuture and @craigham I am on the fence with how we deal with Strings vs Integers when it comes to things like alarm values. ESH gives us the ability to map values in a channel, so we don't have to do a 'mode name' -> 'Mode Number' conversion.

The good thing about this is we keep the binding as simple as possible and use what is built in (and intended).

One the other hand I realize its not as intuitive for users to send '0" as it is to send 'Off', and we do not yet have UI's (except paperui) that populate based on the option labels. This also means users will need a map file if they want nice human readable strings vs ints .

I'm going to get everything working right now with ints, but I would like everyones feedback as it will be easier to switch to strings if needed. I also need to check if ESH has something to solve this, I seem to remember running into this before.